### PR TITLE
feat(Box): Add shorthand props for margin and padding

### PR DESCRIPTION
**Pull request by markdalgleish**
This PR adds the following props to `Box`:

- `padding`
- `paddingX`
- `paddingY`
- `margin`
- `marginX`
- `marginY`

## Migration Guide

Please note that this is entirely optional. The old longhand form will still work

```diff
-<Box paddingLeft="small" paddingRight="small">
+<Box paddingX="small">

-<Box paddingTop="small" paddingBottom="small">
+<Box paddingY="small">

-<Box
-  paddingTop="small"
-  paddingBottom="small"
-  paddingLeft="small"
-  paddingRight="small">
+<Box padding="small">
```

### CODE REVIEW

File: lib/components/private/Field/Field.tsx [hunk 1]
```diff
@@ -37,8 +37,7 @@ interface FieldRenderProps extends Pick<FieldProps, PassthroughProps> {
   boxShadow: BoxProps['boxShadow'];
   borderRadius: BoxProps['borderRadius'];
   width: BoxProps['width'];
-  paddingLeft: BoxProps['paddingLeft'];
-  paddingRight: BoxProps['paddingRight'];
+  paddingX: BoxProps['paddingLeft'];
```
Comment (markdalgleish) [inline lib/components/private/Field/Field.tsx]:
```suggestion
  paddingX: BoxProps['paddingX'];
```

File: lib/components/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -62,14 +62,11 @@ const NamedDropdown = forwardRef<HTMLSelectElement, DropdownProps>(
...
-        {(
-          { className, paddingLeft, paddingRight, ...fieldProps },
-          fieldRef,
-        ) => (
+        {({ className, paddingX, ...fieldProps }, fieldRef) => (
           <Fragment>
             <Box
               component="select"
-              paddingLeft={paddingLeft}
+              paddingLeft={paddingX}
```
Comment (mattsjones) [inline lib/components/Dropdown/Dropdown.tsx]:
```suggestion
              paddingX={paddingX}
```

Comment (markdalgleish) [reply lib/components/Dropdown/Dropdown.tsx]: No, this was actually deliberate. It's applying the horizontal padding, but only on the left side.

Comment (mattsjones) [reply lib/components/Dropdown/Dropdown.tsx]: Weird...

Comment (markdalgleish) [reply lib/components/Dropdown/Dropdown.tsx]: It's because it's applying the right padding to a different element.

**Pull request merged**